Core/PaymentMethodHelper.php

```diff
@@ -70,17 +72,152 @@ public static function getModulePayments()
     }
 
     /**
-     * Generates a Mandate for SEPA transactions
+    * Generates a Mandate for SEPA transactions
+    *
+    * @param int $iOrderNumber
+    *
+    * @return Mandate
+    *
+    * @since 1.0.1
+    */
+    public static function getMandate($iOrderNumber)
+    {
+        $iLength = self::MAX_MANDATE_ID_LENGTH - 1 - strlen((string) time());
+        return new Mandate(substr($iOrderNumber, 0, $iLength) . '-' . time());
+    }
+
+    /**
+     * Returns account holder for Sepa Direct Debit
      *
-     * @param int $iOrderNumber
+     * @return string
      *
-     * @return Mandate
+     * @since 1.0.1
+     */
+    public static function getAccountHolder()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['accountHolder'];
+    }
+
+
+    /**
+     * Returns IBAN
+     *
+     * @return string
      *
      * @since 1.0.1
      */
-    public static function getMandate($iOrderNumber)
+    public static function getIban()
     {
-        $iLength = self::MAX_MANDATE_ID_LENGTH - 1 - strlen((string) time());
-        return new Mandate(substr($iOrderNumber, 0, $iLength) . '-' . time());
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['iban'];
+    }
+
+    /**
+     * Returns BIC
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public function getBic()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['bic'];
+    }
+
+    /**
+     * Generates sepa mandate text
+     *
+     * @param Basket $oBasket
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getSepaMandateHtml($oBasket)
+    {
+        $oShopAddress = Helper::getShopAddress();
+        $iOrderNumber = Helper::getSessionChallenge();
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($oBasket->getPaymentId());
+
+        $sSepaMandateHeader = self::getSepaMandateHeader($oShopAddress, $oPayment, $iOrderNumber);
+        $sSepaMandateFooter = self::getSepaMandateFooter($oShopAddress);
+        $sSepaMandateMain = self::getSepaMandateMainText($oShopAddress);
+
+        if ($oPayment->oxpayments__wdoxidee_sepamandatecustom->value) {
+            $sSepaMandateMain = $oPayment->oxpayments__wdoxidee_sepamandatecustom->value;
```

Comment:
Please ensure that `sepamandatecustom` can only be **plain text** _(discard any HTML tags)_, and then wrap it in a `<p>` tag here.